### ISSUE TITLE
Update notification steals focus

### ISSUE THREAD

**Issue by JosefNemec** [JosefNemec/Playnite#995]
Original report: https://www.reddit.com/r/playnite/comments/aoy9lg/update_notifications_while_running_fullscreen/

Update notification should definitely not steal focus from any application currently in use. Ideally it should open in background or notification should  be delivered via Window's notification bar when Playnite window is not active.

**Comment by Mates1500** [comment]
Would a simple [balloon tip](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.notifyicon.showballoontip) suffice? Enhanced notifications tailored for W10 would be nicer, but could make the OS dependency a little messy.
I propose that the balloon notification could just inform the user of a new version in case the check runs in the background (which is hardcoded for every 6 hours or so iirc?). Then if the user manually uses the update check, it would pop up the actual update window as it does now (therefore not stealing the focus in the background for when you're playing a game)..

**Comment by JosefNemec** [comment]
Ideally we should use Win10 notification system (and the balloon thingy for Win 7/8), because Win10 is by far most used by Playnite users and it has built in feature to suppress notification when game is running.

We also have a notification system built in Playnite, but it's visible only when main window is visible. We would probably need to change the tray icon to signal it in cases when the main window is not visible. That would mean that user will get the info about new update even when system wide notifications are disabled.

Lastly, I've tried to reproduce problem reported by OP and wasn't able to. No matter what I tried (different Playntite UI states and couple of games), I was unable to make Playnite to minimize a game. The window itself doesn't have "on top" flag, so it makes sense it would not take over the main focus. But maybe he uses different Windows version, I only tried it on 10.

**Comment by Mates1500** [comment]
I'm mostly asking because a Windows 10 notification is by default the fallback of a balloon tip. I don't think I've ever seen a "balloon tip" in W10 in my own experience.
From what I've tested, the notifications get blocked while gaming:
 
![aftergame](https://user-images.githubusercontent.com/732314/54073768-478fbb00-428b-11e9-90a7-49d42af1d716.jpg)
and show up as standard notifications otherwise

![normalnotification](https://user-images.githubusercontent.com/732314/54073787-83c31b80-428b-11e9-850d-bb8be39c2e19.jpg)

this is without using any W10-centric API, just the default W10 behavior, at least when you invoke the ShowBalloonTip() function from .NET that I've referenced in the previous post.
